src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -798,31 +893,199 @@ let replaceOutputColumn (annotationTableName:string) (existingOutputColumn: Buil
+
+    let columns = excelTable.columns
+    let outputColumn =
+        columns.items
+        |> Array.ofSeq
+        |> Array.tryFind(fun col -> col.name = outputColumnName)
+
+    if outputColumn.IsSome then
+
+        //Only update the output column when it has a new value
+        if outputColumnName <> newBB.Header.ToString() then
+            excelTable.columns.items.[(int) outputColumn.Value.index].name <- newBB.Header.ToString()
+
+        let warningMsg =
+            if outputColumnName = newBB.Header.ToString() then
+                $"Found existing output column \"{outputColumnName}\". Did not change the column because the new output column is the same \"{excelTable.columns.items.[(int) outputColumn.Value.index].name}\"."
+            else                
+                $"Found existing output column \"{outputColumnName}\". Changed output column to \"{excelTable.columns.items.[(int) outputColumn.Value.index].name}\"."
+
+        let msg = InteropLogging.Msg.create InteropLogging.Warning warningMsg
+
+        let loggingList = [
+            msg
+        ]
+
+        loggingList
+    else
+        failwith "Something went wrong! The update output column is not filled with data! Please report this as a bug to the developers."
+
+/// <summary>
+/// Add a new outputcolumn to an annotation table
+/// </summary>
+/// <param name="excelTable"></param>
+/// <param name="arcTable"></param>
+/// <param name="newBB"></param>
+let addOutputColumn (excelTable:Table) (arcTable:ArcTable) (newBB:CompositeColumn) =
+
+    if arcTable.TryGetOutputColumn().IsSome then
+        failwith "Something went wrong! The add output column is filled with data! Please report this as a bug to the developers."
+
+    else
+
+        let rowCount = arcTable.RowCount + 1
+
+        let newColumn = ExcelHelper.addColumn excelTable (newBB.Header.ToString()) rowCount ""
+        let columnBody = newColumn.getDataBodyRange()
+        // Fit column width to content
+        columnBody.format.autofitColumns()
+
+        let msg = InteropLogging.Msg.create InteropLogging.Info $"Added new output column: {newBB.Header}"
+
+        let loggingList = [
+                msg
+        ]
+
+        loggingList
+
 /// Handle any diverging functionality here. This function is also used to make sure any new building blocks comply to the swate annotation-table definition.
-let addAnnotationBlockHandler (newBB:InsertBuildingBlock) =
+let addAnnotationBlockHandler (newBB:CompositeColumn) =
     Excel.run(fun context ->
         promise {
 
-            let! annotationTableName = getActiveAnnotationTableName context
-            let sheet = context.workbook.worksheets.getActiveWorksheet()
-            let annotationTable = sheet.tables.getItem(annotationTableName)
+            //let activeSheet = context.workbook.worksheets.getActiveWorksheet()
+            //Try to get the name of the currently active sheet
+            let! excelTableName = tryGetActiveAnnotationTableName context
```

Comment:
fail if this isNone, then shadow it with its own .Value

Reply:
fixed